### PR TITLE
[raft]  Add a session proto message for idempotency support

### PR DESCRIPTION
**Pull request by luluz66**
**Related issues**: https://github.com/buildbuddy-io/buildbuddy-internal/issues/3452

### CODE REVIEW

File: proto/raft.proto [hunk 1]
```diff
@@ -311,6 +330,9 @@ message BatchCmdRequest {
 
   // Any post-commit hooks that should be run after a batch is *committed*.
   repeated PostCommitHook post_commit_hooks = 5;
+
+  // Session is used to guarantee idempotency for writes on the state machine.
+  optional Session session = 6;
```
Comment (tylerwilliams) [inline proto/raft.proto:335]: Should this be in a header instead?

File: proto/raft.proto [hunk 2]
```diff
@@ -294,6 +294,25 @@ enum FinalizeOperation {
...
+  // An unique identifier of the client session.
+  // Example: 018fcb2f-cfab-718b-974e-64907df56881
+  bytes id = 1;
+  // Each BatchCmdRequest has its unique <id, index>. The index is
+  // monotonically increasing.
+  // Example: 1717100891000000000
+  uint64 index = 2;
+  // A timestamp when the session will be cleaned up; and when the state machine
+  // can safely clean up the entry.
+  int64 expired_at = 3;
```
Comment (tylerwilliams) [inline proto/raft.proto:309]: s/expired_at/expiry/g

And can we specify units in the comment?

File: proto/raft.proto (hunk 1 above)
Comment (luluz66) [reply proto/raft.proto]: Done.

File: proto/raft.proto (hunk 2 above)
Comment (luluz66) [reply proto/raft.proto]: I renamed to expiry_usec.

**Review by tylerwilliams — APPROVED** [review]

**Pull request merged**